graylog2-server/src/main/java/org/graylog2/indexer/SetIndexReadOnlyJob.java

```diff
@@ -70,9 +77,9 @@ public void execute() {
 
         activityWriter.write(new Activity("Flushed and set <" + index + "> to read-only.", SetIndexReadOnlyJob.class));
 
-        if (!disableIndexOptimization) {
+        if (!indexSet.get().getConfig().indexOptimizationDisabled()) {
             try {
-                systemJobManager.submit(optimizeIndexJobFactory.create(index));
+                systemJobManager.submit(optimizeIndexJobFactory.create(index, indexSet.get().getConfig().indexOptimizationMaxNumSegments()));
```

Comment:
I don't get that warning in my IDE and I am checking that at the top of the method.